server/athenian/api/controllers/goal_controller.py

```diff
@@ -16,11 +22,17 @@ async def get_goal_template(request: AthenianWebRequest, id: int) -> web.Respons
     :type id: int
 
     """
-    try:
+    # TODO: drop support for TEMPLATES_COLLECTION after migration to account owned templates
+    if id in TEMPLATES_COLLECTION:
         template_def = TEMPLATES_COLLECTION[id]
-    except KeyError:
-        return ResponseError(NotFoundError("Template %d was not found." % id)).response
-    model = GoalTemplate(id=id, name=template_def["name"], metric=template_def["metric"])
+        model = GoalTemplate(id=id, name=template_def["name"], metric=template_def["metric"])
+    else:
+        row = await get_goal_template_from_db(id, request.sdb)
```

Comment:
Security: must check that the template ID belongs to the account. Hence it's better to change `get_goal_template_from_db` to always take the account ID parameter and add it as another WHERE condition.

Reply:
security check one line below
We could add it as a WHERE if the URL scheme was different, like `/accounts/{account_id}/goal_templates/{template_id}`
or by fetching "accounts ids for request", but this is the way it's done in get_team controller for instance